### PR TITLE
Enable FP16 on ROCm systems

### PR DESCRIPTION
**Pull request by graham63**

### CODE REVIEW

**Comment by benson31** [comment]
This has been rebased and now builds successfully.

File: include/hydrogen/utils/HalfPrecision.hpp [hunk 1]
```diff
@@ -122,14 +125,9 @@ struct TypeTraits<gpu_half_type>
...
 
-#elif defined(HYDROGEN_HAVE_AMDGPU)
-/** @brief Unified name for the FP16 type on GPU */
-using gpu_half_type = rocblas_half;
-#endif // HYDROGEN_HAVE_CUDA
-
 }// namespace hydrogen
 
-#if defined(HYDROGEN_HAVE_CUDA) && !defined(__CUDACC__)
+#if !(defined(__CUDACC__))// || defined(__HIPCC__))
```
Comment (tbennun) [inline include/hydrogen/utils/HalfPrecision.hpp]: what does this change do?

Comment (benson31) [reply include/hydrogen/utils/HalfPrecision.hpp]: Oh, it's generally wrong. I'll fix it.

**Review by benson31 — APPROVED** [review]

**Review by ndryden — APPROVED** [review]
Looks good to me.

**Pull request merged**